CRM/Contribute/BAO/Contribution.php

```diff
@@ -2008,7 +2008,7 @@ public static function transitionComponents($params, $processContributionObject
               );
             }
 
-            $updateResult['membership_end_date'] = CRM_Utils_Date::customFormat($dates['end_date'],
+            $updateResult['membership_end_date'] = CRM_Utils_Date::customFormat($membership->end_date,
```

Comment:
The fact we have customFormat happening here is a bit alarming & it suggests to me this isn't the right place & we should be loading the actual end date much closer to where the message is sent from - I see this is being called from 2 places - one being transitionComponentWithReturnMessage( - if this is about what appears in the message then that function would be the right place.

Note this function is on our toxic functions list - there used to be a bot that added a link to the toxic code protocol page whenever a PR touched a toxic function - https://wiki.civicrm.org/confluence/display/CRM/Toxic+Code+Protocol

Reply:
thanks @eileenmcnaughton will take a look